src/main/java/net/sf/jabref/gui/openoffice/OOBibBase.java

```diff
@@ -1296,4 +1297,31 @@ public int hashCode() {
 
     }
 
+
+    public BibDatabase generateDatabase(List<BibDatabase> databases, OOBibStyle style)
+            throws NoSuchElementException, WrappedTargetException {
+        BibDatabase resultDatabase = new BibDatabase();
+        List<String> cited = findCitedKeys();
+        for (String key : cited) {
+            for (BibDatabase loopDatabase : databases) {
+                Optional<BibEntry> entry = loopDatabase.getEntryByKey(key);
+                if (entry.isPresent()) {
+                    BibEntry clonedEntry = (BibEntry) entry.get().clone();
+                    clonedEntry.setId(IdGenerator.next());
+                    resultDatabase.insertEntry(clonedEntry);
+                    entry.get().getFieldOptional(FieldName.CROSSREF).ifPresent(crossref -> {
```

Comment:
I added some comments. Most of the time I like Optional (even though I'm at the best is using `ifPresent`), but the lack of `if ... else ...` and the fact that the lambda becomes its own function (which is very natural, but still prohibits writing some things, like return or break) is a bit annoying.